src/tools/OverlayTool.js

```diff
@@ -29,6 +29,56 @@ export default class OverlayTool extends BaseTool {
     this.initialConfiguration = initialConfiguration;
   }
 
+  unpackOverlay(arrayBuffer) {
+    const bitArray = new Uint8Array(arrayBuffer);
+    const byteArray = new Uint8Array(8 * bitArray.length);
+    let nonZero = false;
+
+    for (let byteIndex = 0; byteIndex < byteArray.length; byteIndex++) {
+      const bitIndex = byteIndex % 8;
+      const bitByteIndex = Math.floor(byteIndex / 8);
+      byteArray[byteIndex] =
+        1 * ((bitArray[bitByteIndex] & (1 << bitIndex)) >> bitIndex);
+      if (byteArray[byteIndex]) nonZero = true;
+    }
+
+    return byteArray;
+  }
+
+  /** Gets a byte array of overlay data.
+   * Defaults to returning pixelData if it is a Uint8Array, otherwise tries generating a Uint8Array from:
+   * 1. pixelData.InlineBinary, if it is Base64 encoded, then decodes to Value as a Uint8Array
+   * 2. pixelData.Value, if it is a Uint8Array, assumes it is a packed array
+   * 3. pixelData.retrieveBulkData.then(enabledElement.refresh), assuming it needs to fetch it.
+   * This method does depend on some of the JSON DICOM representations, but those don't have to come from
+   * JSON, it falls back to the original Uint8Array representation, which can come from anywhere.
+   */
+  getOverlayData(pixelData, element) {
```

Comment:
@chafey This is the asynchronous version of the retrieve code.  It assumes 'dicom' format if the retrieveBulkData is present, which isn't unreasonable, and otherwise falls back to the existing behaviour.  To do that, the retrieve call is made, but also the data is handled as InlineBinary and other various formats to support JSON based overlays.